lib/util/include/hse_util/workqueue.h

```diff
@@ -104,4 +105,26 @@ cancel_delayed_work(struct delayed_work *work);
 void
 delayed_work_timer_fn(unsigned long data);
 
+/**
+ * end_stats_work() - Mark the end of a work loop iteration
+ *
+ * This function should be called by long-running workqueue callbacks
+ * which implement a work-processing loop that rarely returns.  Such
+ * callbacks should call end_stats_work() at the end of each loop
+ * iteration to update workqueue statistics.
+ */
+void
+end_stats_work(void);
+
+/**
+ * begin_stats_work() - Mark the beginning of a work loop iteration
+ *
+ * This function should be called by long-running workqueue callbacks
+ * which implement a work-processing loop that rarely returns.  Such
+ * callbacks should call begin_stats_work() at the begging of each
+ * loop iteration to update workqueue statistics.
+ */
+void
+begin_stats_work(void);
```

Comment:
begin_work_loop()/end_work_loop()?

Reply:
Confusing, does end_work_loop() cause the loop to exit?  I'm up for better names, originally I had pause_work() and resume_work() but I thought that was also misleading (does it sleep?)

The reason _work() is at the end is because of the stupid naming of the workqueue API with its giant mess of function names that make it impossible to tell they all come from the same family/subsystem.

Reply:
Sounds good. Workqueue API improvements incoming? 😄

Reply:
Heh... I would like to rename it to something like "threadpool" and unify the API naming :-)

Reply:
Makes more sense to me that way